Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+{
+    mavlink_encoding_status_t encoding_status = MAVLINK_ENCODING_INCOMPLETE;
+
+    uint8_t system_id = 1;
+    uint8_t component_id = 1;
+
+    mavlink_message_t encoded_msg_original;
+    memset(&encoded_msg_original, 0, sizeof(mavlink_message_t));
+
+    uint16_t message_len;
+
+    mavlink_global_position_int_t global_position;
+
+    switch(msgID)
+    {
+        case MESSAGE_ID_TIMESTAMP:
+        {
+            POGI_Timestamp_t* timestamp_cmd = (POGI_Timestamp_t*) struct_ptr;
+
+            global_position.lat = timestamp_cmd ->timeStamp;
+        } 
+        break;
+
+        case MESSAGE_ID_GPS:
+        {
+            POGI_GPS_t* warg_GPS_cmd = (POGI_GPS_t*) struct_ptr;
+
+            global_position.lat = warg_GPS_cmd ->latitude;
+            global_position.lon = warg_GPS_cmd ->longitude;
+            global_position.alt = warg_GPS_cmd ->altitude;
+        } 
+        break;
+
+        case MESSAGE_ID_EULER_ANGLE_PLANE:
+        case MESSAGE_ID_EULER_ANGLE_CAM:
+        {
+            POGI_Euler_Angle_t* gimbal_cmd = (POGI_Euler_Angle_t*) struct_ptr;
+
+            global_position.lat = gimbal_cmd ->roll;
+            global_position.lon = gimbal_cmd ->pitch;
+            global_position.alt = gimbal_cmd ->yaw;
+        }
+        break;
+
+        case MESSAGE_ID_IS_LANDED:
+        case MESSAGE_ID_HOMEBASE_INITIALIZED:
+        {
+            single_bool_cmd_t* warg_cmd = (single_bool_cmd_t*) struct_ptr;
+
+            global_position.hdg = warg_cmd ->cmd;
+        }
+        break;
+
+        case MESSAGE_ID_AIR_SPEED:
+        case MESSAGE_ID_CURRENT_WAYPOINT_LD:
+        case MESSAGE_ID_CURRENT_WAYPOINT_INDEX:
+        {
+            four_bytes_int_cmd_t* numWaypoint_cmd = (four_bytes_int_cmd_t*) struct_ptr;
+
+            global_position.lat = numWaypoint_cmd ->cmd;
+        } 
+        break;
+
+        case MESSAGE_ID_ERROR_CODE:
+        case MESSAGE_ID_EDITING_FLIGHT_PATH_ERROR_CODE:
+        case MESSAGE_ID_FLIGHT_PATH_FOLLOWING_ERROR_CODE:
+        {
+            one_byte_uint_cmd_t* warg_cmd = (one_byte_uint_cmd_t*) struct_ptr;
+
+            global_position.hdg = warg_cmd ->cmd;
+        }
+        break;
+
+        default:
+            return MAVLINK_ENCODING_BAD_ID;
+            break;
+    }
+
+    // loading warg command ID before encoding
+    global_position.time_boot_ms = msgID;
+    message_len = mavlink_msg_global_position_int_encode(system_id, component_id, &encoded_msg_original, &global_position);
+
+    if (message_len == 0)
+    {
+        return MAVLINK_ENCODING_FAIL;
+    }
+
+    // the following loop is supposed to move the two checksum bytes to the last so that the encoder 
+    // can send out a message (byte array) that starts exactly with the starting byte
+    unsigned char* ptr_in_byte = (unsigned char *) &encoded_msg_original;
+    char message_buffer[message_len];
```

Comment:
I feel like this would require heap-allocation. Normally compilers don't let you declare arrays with variable lengths on the stack

Reply:
I changed this to 39 which should be the max length needed for our purpose